tests/nvmeof/test_ceph_nvmeof_neg_tests.py

```diff
@@ -500,365 +608,102 @@ def test_ceph_83576087(ceph_cluster, rbd, pool, config):
-        client.exec_command(sudo=True, cmd=f"mount {targets[0]} {_dir}")
-        client.exec_command(sudo=True, cmd=f"cp /var/log/messages {_file}")
-        client.exec_command(sudo=True, cmd=f"ls -ltrh {_file}")
-
-        # Reboot NVMeoF GW node and wait for the node recovery.
-        # Wait for the GW service to be up and running.
-        if not reboot_node(gw_node):
-            raise Exception("Host did not started post reboot!!!!!")
-
-        service_name = f"nvmeof.{pool}"
-        gw_group = config.get("gw_group")
-        service_name = f"{service_name}.{gw_group}" if gw_group else service_name
-        check_service_exists(
-            ceph_cluster.get_nodes(role="installer")[0],
-            service_name=service_name,
-            service_type="nvmeof",
-        )
-        client.exec_command(sudo=True, cmd=f"ls -ltrh {_file}")
-        client.exec_command(sudo=True, cmd=f"cp /var/log/messages {_file}_test")
-        client.exec_command(sudo=True, cmd=f"ls -ltrh {_file}_test")
-        client.exec_command(sudo=True, cmd=f"umount {_dir}")
-        LOG.info("Validation of CEPH-83576093 is successful.")
-    except Exception as err:
-        raise Exception(err)
-    finally:
-        cleanup_cfg = {
-            "gw_node": config["gw_node"],
-            "disconnect_all": [config["initiator_node"]],
-            "cleanup": ["disconnect_all", "gateway", "pool"],
-            "rbd_pool": pool,
-        }
-        config.update(cleanup_cfg)
-        teardown(ceph_cluster, rbd, nvmegwcli, config)
-
-
-def test_ceph_83575455(ceph_cluster, rbd, pool, config):
-    """CEPH-83575455: Validate Host access failures"""
-    gw_node = get_node_by_id(ceph_cluster, config["gw_node"])
-    ceph = Orch(ceph_cluster, **{})
-    nvmegwcli = create_gateway(
-        nvme_gw_cli_version_adapter(ceph_cluster),
-        gw_node,
-        mtls=config.get("mtls"),
-        shell=getattr(ceph, "shell"),
-        port=config.get("gw_port", 5500),
-        gw_group=config.get("gw_group"),
-    )
-    client = get_node_by_id(ceph_cluster, config["initiator_node"])
-    initiator = NVMeInitiator(client)
-    initiator_nqn = initiator.initiator_nqn()
-
-    subsystem = dict()
-    listener_port = find_free_port(gw_node)
-    subsystem.update(
-        {
-            "nqn": "nqn.2016-06.io.spdk:ceph-83575455",
-            "serial": 113,
-            "listener_port": listener_port,
-            "allow_host": initiator_nqn,
-        }
-    )
-    initiator_cfg = {
-        "subnqn": "nqn.2016-06.io.spdk:ceph-83575455",
-        "listener_port": listener_port,
-        "node": config["initiator_node"],
-    }
-
-    _dir = f"/tmp/dir_{generate_unique_id(4)}"
-    _file = f"{_dir}/test.log"
-
-    try:
-        configure_subsystems(ceph_cluster, rbd, pool, nvmegwcli, subsystem)
-        name = generate_unique_id(length=4)
+        # Add host
+        configure_hosts(nvme_service.gateways[0], nvme_service.config)
```

Comment:
Can we not directly call configure_gw_entities and that will take care of configuring subsystems, listeners and hosts right? All you need to do is pass those keywords to config I guess. Can we check this?